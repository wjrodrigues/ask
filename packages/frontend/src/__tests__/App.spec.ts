import { describe, it, expect } from "vitest";
import { mount, shallowMount } from "@vue/test-utils";

import AppVue from "@/App.vue";
import { basic_mount } from "./help";
import { createRouter, createWebHistory } from "vue-router";

describe("when render AppVue", () => {
  it("renders route component with title", async () => {
    const router = createRouter({
      history: createWebHistory(),
      routes: [
        {
          path: "/",
          component: { template: "Welcome" },
          meta: { title: "signup.title" },
        },
      ],
    });

    const wrapper = mount(
      AppVue,
      basic_mount({ props: {}, plugins: [router] })
    );

    router.push("/");
    await router.isReady();

    expect(wrapper.text()).toEqual("Welcome");
    expect(wrapper.vm.$el.ownerDocument.title).toEqual("Inscrever-se | Ask");
  });

  it("renders route component without title", async () => {
    const router = createRouter({
      history: createWebHistory(),
      routes: [
        {
          path: "/",
          component: { template: "Welcome" },
          meta: {},
        },
      ],
    });

    const wrapper = mount(
      AppVue,
      basic_mount({ props: {}, plugins: [router] })
    );

    router.push("/");
    await router.isReady();

    expect(wrapper.text()).toEqual("Welcome");
    expect(wrapper.vm.$el.ownerDocument.title).toEqual("Ask");
  });

  it("match snapshot", async () => {
    const router = createRouter({
      history: createWebHistory(),
      routes: [
        {
          path: "",
          component: { template: "Welcome" },
          meta: {},
        },
      ],
    });

    const wrapper = shallowMount(
      AppVue,
      basic_mount({ props: {}, plugins: [router] })
    );

    expect(wrapper.html()).toMatchSnapshot();
  });
});
