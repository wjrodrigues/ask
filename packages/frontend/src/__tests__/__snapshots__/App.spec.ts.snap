// Vitest Snapshot v1

exports[`when render AppVue > match snapshot 1`] = `
"<router-view-stub name=\\"default\\"></router-view-stub>
<snack-bar-stub></snack-bar-stub>"
`;
